spec/features/cdr/cdr_history/cdr_filter_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+RSpec.describe 'Cdrs index page filtering', js: true do
+  subject do
+    visit cdrs_path
+    filter_cdr_records
+    within_filters { click_submit('Filter') }
+  end
+
+  before do
+    Cdr::Cdr.delete_all
+  end
+
+  include_context :login_as_admin
+
+  let(:filter_value) { nil }
+
+  let!(:tags) { create_list(:routing_tag, 4) }
+  let!(:cdrs_not_tagged) do
+    [
+      create(:cdr, routing_tag_ids: []),
+      create(:cdr, routing_tag_ids: nil)
+    ]
+  end
+  let!(:cdrs_tagged) do
+    [
+      create(:cdr, routing_tag_ids: [tags.first.id]),
+      create(:cdr, routing_tag_ids: [tags.second.id, tags.third.id]),
+      create(:cdr, routing_tag_ids: [tags.first.id, tags.fourth.id])
+    ]
+  end
+
+  describe 'filter by tagged' do
+    let(:filter_cdr_records) do
+      within_filters do
+        fill_in_chosen 'Tagged', with: filter_value
+      end
+    end
+
+    context 'with filter by tagged' do
+      let(:filter_value) { 'Yes' }
+
+      it 'shows filtered records with routing_tags' do
+        subject
+        expect(page).to have_table
+        expect(page).to have_table_row count: cdrs_tagged.size
+        cdrs_tagged.each { |cdr| expect(page).to have_table_cell column: 'Id', text: cdr.id }
+        expect(page).to have_field_chosen('Tagged', with: 'Yes')
```

Comment:
wrap this line in `within_filters do .. end`

it will be easier to understand that you check filter value here, not just value is some form

Reply:
same for below tests